fuel-merkle/test-helpers/src/binary/verify.rs

```diff
@@ -1,42 +1,44 @@
 use crate::binary::{
```

Comment:
I think that reusing the function here is worth considering. 
As part of that consideration, some factors include: 
- Do we have a production use case for `fuel_merkle::binary::verify` yet? If not, it might actually make more sense to keep only `test-helpers::binary::verify` instead. `test-helpers::binary::verify` is used specifically to verify proofs during tests (i.e. check the proofs are generated correctly), but `fuel_merkle::binary::verify` doesn't have a "real" use case, so it is currently just a maintenance burden.
- Can we simplify or improve the implementation of `fuel_merkle::binary::verify`? If so, we can use `test-helpers::binary::verify` to quickly check the correctness of any refactors to the production version.